lcviz/plugins/ephemeris/ephemeris.py

```diff
@@ -214,10 +214,9 @@ def _update_all_phase_arrays(self, *args, component=None):
                 )
 
                 new_links.append(new_link)
-                dc.add_link(new_link)
 
-        if len(new_links):
-            dc.set_links(new_links)
+        for new_link in new_links:
+            dc.add_link(new_link)
```

Comment:
Despite the name, this accepts a single link or list of links

```suggestion
        dc.add_link(new_links)
```

Reply:
(I'm just going to commit this so I can test with it... I don't think it'll be controversial 🤞 )